spec/tapioca/cli/dsl_spec.rb

```diff
@@ -343,6 +343,62 @@ def title=(title); end
           assert_success_status(result)
         end
 
+        it "generates RBI files for engine when provided with an `app_root` flag" do
+          @project.write("lib/post.rb", <<~RB)
+            require "smart_properties"
+
+            class Post
+              include SmartProperties
+              property :title, accepts: String
+            end
+          RB
+
+          @project.write("lib/comment.rb", <<~RB)
+            require "smart_properties"
+
+            module Namespace
+              class Comment
+                include SmartProperties
+                property! :body, accepts: String
+              end
+            end
+          RB
+
+          engine_path = @project.path + "/test/dummy"
+          FileUtils.mkdir_p(engine_path)
+          FileUtils.mv(@project.path + "/config", engine_path)
+          FileUtils.mv(@project.path + "/lib", engine_path)
+
+          result = @project.tapioca("dsl --app_root=test/dummy")
+
+          assert_empty_stderr(result)
+
+          assert_project_file_equal("sorbet/rbi/dsl/post.rbi", <<~RBI)
+            # typed: true
+
+            # DO NOT EDIT MANUALLY
+            # This is an autogenerated file for dynamic methods in `Post`.
+            # Please instead update this file by running `bin/tapioca dsl Post`.
+
+            class Post
+              include SmartPropertiesGeneratedMethods
+
+              module SmartPropertiesGeneratedMethods
+                sig { returns(T.nilable(::String)) }
+                def title; end
+
+                sig { params(title: T.nilable(::String)).returns(T.nilable(::String)) }
+                def title=(title); end
+              end
+            end
+          RBI
+
+          # Restore directory structure so to not impact other tests
+          FileUtils.mv(engine_path + "/config", @project.path)
```

Comment:
Can we put this inside an `ensure` block? It makes sure that we restore back even if something fails.

Reply:
Good call, I've pushed an update.